### ISSUE TITLE
One report file generated when running multiple suites

### ISSUE THREAD

**Issue by danielezer** [cawolf/behat-cucumber-formatter#4]
Hi, 
I am using this package (which is great) and running multiple test suites in different directories.
When running the command:
behat --config behat.yml -f pretty -o std -f junit -o reports -f cucumber_json -o reports

After the command has finished I only get one report file by the name of the last suite.

I have this in my behat.yml file:
      Vanare\BehatCucumberJsonFormatter\Extension:
        fileNamePrefix: report-
        outputDir: %paths.base%/reports

Can you please help me see why I only get one file?
Thanks!

**Comment by 8p8c** [comment]
Isn't it a correct behavior? You should get only one JSON report for all of your tests, because you can (and must) feed only one JSON report to your Jenkins Cucumber plugin, which builds beautiful pages for all of your tests.

**Comment by danielezer** [comment]
Hi,
Thank you for the quick response!
What I meant is that I am getting one report of only the last suite which includes the suite’s name in the file.

**Comment by 8p8c** [comment]
Sorry, missed that thing.
I'm not a Behat specialist and never was, but is it legal in your country to launch few test suites at once? I mean do you really launch several test suites, do you see those tests running in your console output?

**Comment by danielezer** [comment]
Lol, when I run all the suites, I can see a summary of all the tests in stdout. If I run each suite separately I will only see the summary of each suite at the end of that’s suite run which will be hard and uncomfortable to find in the jenkins console output, when I am really interested in the summary.

**Comment by danielezer** [comment]
Also, it is much more comfortable to run one command in order to run the tests instead of one command for each suite (or a loop). Also, from what you said, I understand that I can only feed the Jenkins cucumber plugin with one report file so anyway a number of files with different names are not good, right?

**Comment by 8p8c** [comment]
I'm not sure Jenkins Cucumber plugin could work with several test suites. Does original ruby cucumber supports test suites? Is there a way to put several test suites in a single report.json? Can you point me to an example of a such report? If I am right, then behat-cucumber-formatter would not create a report with several test suites.

**Comment by danielezer** [comment]
Hi, 
The report used to work in past with the Vanare/behat-cucumber-formatter but I started to get errors in the cucumber reports so I tried using your formatter. It actually did help but I am only seeing results of one suite. I don't have an example report file of multiple suites since the report haven't been working for quite some time. I see that the original plugin does have support for multiple suites somehow - https://github.com/Vanare/behat-cucumber-formatter/issues/12 so I guess it is possible to work with multiple suites

**Comment by 8p8c** [comment]
Can you show me an example of multi-suite report made by cucumber? Can you give me a link to cucumber documentation that describes multiple suites? How Jenkins should render it's pages for multiple suites? I'm still not sure cucumber is able to use multiple suites or run few of them at once.

**Comment by cawolf** [comment]
Hey, sorry for the delay.

Actually, as I forked the project primarily for our use case, the single suite approach is a feature. We are using a plugin that is capable of using one (and exactly one) input JSON file, and we are not fond of merging multiple suite JSONs into a single one, although that would actually be possible. It is, however imho beyond the scope of this formatter.

**Comment by cawolf** [comment]
No further feedback provided, closing.

**Comment by grEvenX** [comment]
I'm running into the same problem. We split our tests into multiple suites to not make huge and complex FeatureContext files that contains the step definitions for our entire application.

It's quite common to split the results per feature-file (which was the original implementation as far as I can tell), as can also be seen by Behat's built-in junit formatter. 
The Jenkins plugin also supports multiple JSON files to combine them in a report as per their documentation:
```
node {
    stage('Generate HTML report') {
        cucumber buildStatus: 'UNSTABLE',
                reportTitle: 'My report',
                fileIncludePattern: '**/*.json',
                trendsLimit: 10,
                classifications: [
                    [
                        'key': 'Browser',
                        'value': 'Firefox'
                    ]
                ]
    }
}
```

Any chance of this getting implemented in this repo from a PR, or is another fork preferred?

**Comment by cawolf** [comment]
@grEvenX you are welcome to provide a PR for this feature, we do not have the former restrictions from our processing pipeline and therefore no reason to keep just one result file.

However, if it is possible to easily configure the behaviour from the outside, I would prefer a solution that provides both ways.

**Comment by grEvenX** [comment]
@cawolf Created a PR as a starting point: https://github.com/cawolf/behat-cucumber-formatter/pull/7

**Comment by cawolf** [comment]
Released in [v1.4.0](https://github.com/cawolf/behat-cucumber-formatter/releases/tag/v1.4.0)
